paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java

```diff
@@ -70,6 +73,13 @@ public abstract class FileDeletionBase {
 
     protected final Map<BinaryRow, Set<Integer>> deletionBuckets;
     protected final Executor ioExecutor;
+    protected boolean changelogDecoupled;
```

Comment:
changelogDecoupled who use this?

Reply:
For example, `SnapshotDeletion` will use this to determine whether to clean changelog manifests. It will also be used in `ChangelogDeletion` in following PR.

Reply:
but there is no assignment for it.

maybe you need to add it in next pr.

Reply:
There is `setChangelogDecoupled` . I feel difficult to make this field `final`, so I add a set method for it.